common/lib/dependabot/clients/codecommit.rb

```diff
@@ -193,6 +196,19 @@ def create_pull_request(pr_name, target_branch, source_branch,
         )
       end
 
+      def truncate_pr_description(pr_description)
+        # Codecommit only supports descriptions up to 10240 chars in length
+        # https://docs.aws.amazon.com/codecommit/latest/APIReference/API_PullRequest.html
+        pr_description = pr_description.to_s
+        if pr_description.length > MAX_PR_DESCRIPTION_LENGTH
+          truncated_msg = "...\n\n_Description has been truncated_"
+          truncate_length = MAX_PR_DESCRIPTION_LENGTH - truncated_msg.length
+          pr_description = (pr_description[0..truncate_length] + truncated_msg)
```

Comment:
This whole section of var names + logic is really clean/readable, well done. 💯

Reply:
Oh, haha, this is copy/pasted from Azure... kudos to @jurre on the cleanness of his original implementation.

Reply:
And this also exists here too: https://github.com/dependabot/dependabot-core/blob/221a6e5599ea1e8c30f6167b0a1bbe48c0b8e15f/common/lib/dependabot/pull_request_creator/github.rb#L351-L362

oof, this will get out of hand quickly.

Reply:
Could extract a little helper? Although these are unlikely to ever change